src/Proxy.hs

```diff
@@ -890,5 +893,47 @@ putGTUDropR addrText = do
 getGlobalFileR :: Handler TypedContent
 getGlobalFileR = toTypedContent . globalInfo <$> getYesod
 
+-- Queries the transaction database and the GRPC, 
+-- then if both succeed checks that the last final block is less than 5 minutes old.
+getHealthR :: Handler TypedContent
+getHealthR = 
+  runGRPC doGetBlockFinalInfo $ \case
+      Nothing -> do
+        $(logError) $ "Could not get response from GRPC."
+        sendResponse $ object $ ["healthy" .= False, "reason" .= ("Could not get response from GRPC.":: String)]
+      Just lastFinalBlockInfo -> do
+        $(logInfo) "Successfully got best block info."
+        result :: [(Entity Entry, Entity Summary)] <- runDB $ E.select $
+                  E.from $ \val -> do
+                  E.limit 1 -- query for any single row
+                  return val
+        case result of
+          [] ->
```

Comment:
Why is no result considered an error? What if the table is empty?

Reply:
Instead I'd recommend just making the query. If the database is not reachable an exception will be thrown, or a timeout will occur.